### DISCUSSION TITLE
Query About Hacktoberfest

### DISCUSSION THREAD

**Discussion by Harshil-Jani** [Tenet-Coding/ArtofCoding#41] [Q&A]
![image](https://user-images.githubusercontent.com/79367883/133957840-00a78942-58b0-4a19-a732-87e0d4affb8c.png)

### Hey, Hacktoberfest is on it's way. So are we open for Hacktoberfest contributions ? If yes, Then we can plan something more attractive apart from only codes which will help in increasing Quality Contributions ! Any Suggestions ?

**Comment by soumikbaithalu** [comment]
Yay @Harshil-Jani, I actually want to talk regarding  hacktoberfest2021  how should be engaging the new user so that more contribute to our repo.

**Reply by soumikbaithalu** [reply]
@Harshil-Jani  I think I should remove the restyle.io so that contributer need to write their own clean, not the bot ??

**Reply by Harshil-Jani** [reply]
Yes ! I also feel so.

**Reply by Harshil-Jani** [reply]
> Yay @Harshil-Jani, I actually want to talk regarding hacktoberfest2021 how should be engaging the new user so that more contribute to our repo.

I would suggest that, Let's make this open as a project repo ! People will Create Issues in which they need to describe about their project in more details, then we can assign them to do it. Also, Some project issues we will create and assign whoever asks. It will be in JS, Python, HTML,CSS Anything.

We will ask them to name folders and give some sort of numbering as the projects arrive by renaming issue.

**Reply by soumikbaithalu** [reply]
That great idea @Harshil-Jani you can create a new repo for all the small and medium and large projects.

**Reply by Harshil-Jani** [reply]
@soumik2012 Shall I create the repo in Tenet Coding named as Hacktoberfest-Projects ? And do some work on README and other files.

**Reply by soumikbaithalu** [reply]
Yay @Harshil-Jani create a new repo in tenet coding also you should add your small project for the demo purpose.

**Reply by soumikbaithalu** [reply]
Hey @Harshil-Jani  create some issues so that beginner will easy solve the problem

**Reply by Harshil-Jani** [reply]
@soumik2012 Hey! We should not merge any PR before October 1st right? Otherwise it won't be counted as Hacktoberfest contribution.

**Reply by soumikbaithalu** [reply]
Yup @Harshil-Jani

**Reply by soumikbaithalu** [reply]
@Harshil-Jani One of the contributor make a PR he is not assigned to the problem what should be do now??

**Reply by Harshil-Jani** [reply]
If someone makes the PR without asking. In my opinion.. If there is no one who asked it before him, then we should merge it and guide them to ask before making PR so they won't waste time in future Open Source Journey. Else if someone already asked for the issue before the PR was made then let the PR be pending then we shall compare both the PR and merge one of them and ask other to improve their work.

**Reply by soumikbaithalu** [reply]
@Harshil-Jani okay..

**Reply by soumikbaithalu** [reply]
@Harshil-Jani Hackotberfest is now open should we accept the PR or not ??

**Reply by soumikbaithalu** [reply]
@Harshil-Jani  https://hacktoberfest.digitalocean.com/resources/maintainers please go through this.

**Reply by Harshil-Jani** [reply]
> @Harshil-Jani Hackotberfest is now open should we accept the PR or not ??

Nope...Anything before Oct 1 is not a part of Hacktoberfest. Current Phase of 2 days is only for registrations. Event has not yet began.
![image](https://user-images.githubusercontent.com/79367883/135261261-11092d4f-4f9e-4b20-b852-01c04a90a1cc.png)

https://hacktoberfest.digitalocean.com/faq

**Reply by Harshil-Jani** [reply]
> @Harshil-Jani https://hacktoberfest.digitalocean.com/resources/maintainers please go through this.

Sure :100:

**Reply by Harshil-Jani** [reply]
@soumik2012 
Once the contribution starts, Shall we keep time limits on issues ? 
For Example :
Easy marked issues - If the contributor don't give updates (Not completing the whole project but convincing us that he/she is still working for this issue) within 2 days or so, We can assign it to someone else ?
Similarly for Medium level 5 days and so on.
What are your views on this ?

**Reply by Harshil-Jani** [reply]
Also Shall I create some Templates for Issues and PRs ? So that it will be easy for contributors to create issues.

**Reply by soumikbaithalu** [reply]
1. Okay, @Harshil-Jani actually I have given 7-10 days to complete the task but Hacktoberfest 1 month so your idea is nice we will implement this.
2. Actually I am going to do that but it is okay.. you can take it. You know na what folder we have to put **Issue and PR templates**
   if you don't reach out to me I will guide you.

**Reply by Harshil-Jani** [reply]
I have never made Issues and PR Templates but looking at Art Of Coding Repository, What I understood is we need to store these template files into .github folder. Is this correct ? If I am missing something please guide me.

> 1. Okay, @Harshil-Jani actually I have given 7-10 days to complete the task but Hacktoberfest 1 month so your idea is nice we will implement this.
> 2. Actually I am going to do that but it is okay.. you can take it. You know na what folder we have to put **Issue and PR templates**
>    if you don't reach out to me I will guide you.

**Reply by soumikbaithalu** [reply]
@Harshil-Jani First Go to GitHub page then go to the insight of our project and go to the community you will see the option like here
![image](https://user-images.githubusercontent.com/66599363/135398871-803a4636-83c0-4d4b-8150-0a96f39d9a23.png)
 Add here and issue template here.

**Reply by soumikbaithalu** [reply]
Also, @Harshil-Jani if check the project tell them to add readme.md files also 

- Name of Project
- Demo of Project in .gif how your project is look 
- Description of project 
-  Requirements or Setup instructions
-  Name of Author(s)  
- [Look Here](https://github.com/Tenet-Coding/Hacktoberfest-Projects/blob/master/ConwaysGameOfLife/README.md)

**Reply by Harshil-Jani** [reply]
@soumik2012 Okay.. I will do required changes by Tonight ! :100: 
Thanks for Guiding regarding Templates

**Reply by soumikbaithalu** [reply]
![image](https://user-images.githubusercontent.com/66599363/135420220-7c483c2c-326d-4a1b-8e7d-dc8c9ba77610.png)
@Harshil-Jani  I have seen you don't show your profile to the public, go to @tenet-coding and go to people change profile to the public it is easy to recognize the people.

**Reply by Harshil-Jani** [reply]
> ![image](https://user-images.githubusercontent.com/66599363/135420220-7c483c2c-326d-4a1b-8e7d-dc8c9ba77610.png) @Harshil-Jani I have seen you don't show your profile to the public, go to @Tenet-Coding and go to people change profile to the public it is easy to recognize the people.

Oh.. Okay ! I am doing it. Thanks for letting me know !

**Comment by Harshil-Jani** [comment]
@soumik2012 I have created the PR and Issue Templates in Hacktoberfest. Check once and if any changes ae required Let me know !

**Reply by Harshil-Jani** [reply]
@soumik2012 I have added ```hacktoberfest-accepted``` label in Hacktoberfest-Project ! So, add this label while you merge any pull request as per maintainers guidelines in Hacktoberfest official site.

**Reply by Harshil-Jani** [reply]
@soumik2012  In hacktoberfest we are asking contributors to upload their projects in new branches and then merge them to our master branch later on right ? Then with [Pull Request 33](https://github.com/Tenet-Coding/Hacktoberfest-Projects/pull/33) the thing is it is commited on master branch of contributor's fork. If we will merge this  then it will affect our master branch as well. So, Shall we ask for commits on a new branch !

**Reply by soumikbaithalu** [reply]
> @soumik2012 I have added `hacktoberfest-accepted` label in Hacktoberfest-Project ! So, add this label while you merge any pull request as per maintainers guidelines in Hacktoberfest official site.
Yup.

**Reply by Harshil-Jani** [reply]
@soumikbaithalu Shall We close the contributions on Hacktoberfest with a Thanking Note to all the contributors ? Or Let it be open as it is ?

**Reply by soumikbaithalu** [reply]
Let it be open but an add Thank you note to Hacktoberfest. @Harshil-Jani
